src/core/contract.service.ts

```diff
@@ -119,31 +121,15 @@ export class ContractService {
     return key.startsWith('0x') && isValidPrivate(toBuffer(key));
   }
 
-  async indexerToController(indexer: string) {
-    const controller = await this.sdk.indexerRegistry.getController(indexer);
-    return controller ? controller.toLowerCase() : '';
-  }
-
   updateSDK(key: string) {
     const keyBuffer = toBuffer(key);
     this.wallet = new Wallet(keyBuffer, this.provider);
     this.sdk = initContractSDK(this.wallet, this.chainID);
   }
 
   async updateContractSDK(indexer: string): Promise<ContractSDK | undefined> {
```

Comment:
```suggestion
  async updateContractSDK(controller: string): Promise<ContractSDK | undefined> {
```

Reply:
The place that calls `updateContractSDK` is in Network Service, we have account service injected there.